### ISSUE TITLE
Add ability to detect memory usage in bin/start

### ISSUE THREAD

**Issue by CrazyICT** [markshust/docker-magento#527]
I started a clean setup, but the elasticsearch container gives some issues
it starts but after a few seconds the containers just stops, without any logging etc.

i tried the both: 

markoshust/magento-elasticsearch:7.71-0 
markoshust/magento-elasticsearch:7.7

(MacOs big sur 11.5.2, docker engine 20.10.8)

**Comment by markshust** [comment]
Is anyone else able to replicate this? I'm currently running this project on a fairly underpowered MacBook Air, and it's running properly on a fresh install. Perhaps you can up the memory/RAM allocated to Docker Desktop and try again? I recommend at least 6GB of RAM allocated to Docker.

**Comment by bakerml** [comment]
Yes - I am experiencing this issue on a MacBook Pro with the M1 chip. I think there is a pull request that references a similar issue that suggests creating an image for elasticsearch 7.9

**Comment by ponickkhan** [comment]
I had faced this issue few weeks ago.  Recent Docker update reset the docker resource settings. By default docker set the memory limit to 2GB which is not enough for elastic search to function proper. After increasing the memory it is just working fine.

**Comment by markshust** [comment]
I just faced this issue myself! @ponickkhan nailed it -- the default RAM for Docker Desktop is set to 2GB. Increase it to at least 6GB.

This caused my so much aggravation, I think within the `bin/start` script we should add a checker to make sure the user has enough RAM allocated.

I found out the RAM usage is defined on the hypervisor, not Docker level. It seems to be output from the `-m` flag of running:

```
ps aux|grep hyperkit
```

![Screen Shot 2021-09-21 at 11 36 15 AM](https://user-images.githubusercontent.com/437029/134201739-8628a2a8-88fc-4208-a9bb-5667d1ff1e84.png)

If someone wants to tackle this, would most likely need some regex here, pull out the value, and make sure it is about 6000 (6GB).

**Comment by markshust** [comment]
@rangerz awesome, thanks! This is much cleaner.

**Comment by markshust** [comment]
@rangerz again, appreciate that line! That made everything easy. All added & committed 👍

**Comment by sethIam1** [comment]
I started the one line curl command and didn't realize I had allocated 5.97GB. I think it stopped midway, it looks like it downloaded and installed Magento, and then I got this message:

There must be at least 6GB of RAM allocated to Docker in order to continue.

Anyone know what do to? I already set the new allocated amount to 8GB, do I just rewrite and start over with the same curl or is there a way to continue the stopped operation?
